cpp/memilio/utils/time_series.h

```diff
@@ -524,13 +527,38 @@ class TimeSeries
             set_ostream_format(out, w, p) << std::right << get_time(i);
             auto res_i = get_value(i);
             for (size_t j = 0; j < static_cast<size_t>(res_i.size()); j++) {
-                out << " ";
+                out << separator;
                 set_ostream_format(out, w, p) << std::right << res_i[j];
             }
         }
         out << "\n";
     }
 
+    /**
+     * @brief Exports a TimeSeries object into a CSV file.
+     *
+     * The first column of the CSV file contains the time points. The remaining columns
+     * contain the values at each time point. Column headers can be specified with column_labels.
+     *
+     * @param filename Path to the CSV file
+     * @param column_labels [Default: {}] Vector of labels for each column after the time column.
+     * @param separator [Default: ','] Separator character.
+     * @param precision [Default: 6] Number of decimals for floating point values.
+     * @return bool Indicates success or failure.
+     */
+    bool export_csv(const std::string& filename, const std::vector<std::string>& column_labels = {},
+                    char separator = ',', int precision = 6) const
+    {
+        std::ofstream file(filename);
+        if (!file.is_open()) {
+            return false;
+        }
+
+        // Use print_table to write the CSV file with width=1, "," separator, and no header prefix
+        print_table(column_labels, 1, precision, file, separator, "");
```

Comment:
Maybe add the comment to the docstring? Could be useful information, maybe.
Also, have you tried importing this somewhere? I think the prefix should be a "#", so that the first line is correctly read as titles (or skipped). I am not sure if the fallback numbering with "#1" etc works well with this, we could change it to eg. "C1" for compartment or column 1.